core/src/main/scala/filodb.core/memstore/OnDemandPagingShard.scala

```diff
@@ -104,7 +104,9 @@ TimeSeriesShard(dataset, storeConfig, shardNum, bufferMemoryManager, rawStore, m
             rawStore.readRawPartitions(dataset, allDataCols, multiPart, computeBoundingMethod(pagingMethods))
               // NOTE: this executes the partMaker single threaded.  Needed for now due to concurrency constraints.
               // In the future optimize this if needed.
-              .mapAsync { rawPart => partitionMaker.populateRawChunks(rawPart).executeOn(singleThreadPool) }
+              .mapAsync { rawPart =>
+                partitionMaker.populateRawChunks(rawPart).executeOn(singleThreadPool).asyncBoundary
+              }
               .doOnTerminate(ex => span.finish())
               // This is needed so future computations happen in a different thread
               .asyncBoundary(strategy)
```

Comment:
Is this needed anymore? Should the strategy be passed to the new async calls too?

Reply:
@velvia It is needed. The locks take care of chunk-map operations.
Partition creation modifies the shard state, namely PartitionSet, PartitionMap etc which need to be altered by ingestion thread only (unless we add additional locking).

Reply:
@velvia can you comment here? Looks like there is an `.asyncBoundary(strategy)` for multiPart == true. Will adding this when multiPart == false also solve the issue ?

Reply:
Can we remove the `.asyncBoundary` after `.executeOn(...)`  ?

Reply:
In theory should be equivalent to the earlier `.asyncBoundary`.  IE, you should need just one `asyncBoundary`, not two like the earlier code.  AsyncBoundaries force thread switches btw if the thread pool has multiple threads or you are scheduling executions on a different pool/scheduler.